### DISCUSSION TITLE
How to specify push (P) target remote?

### DISCUSSION THREAD

**Discussion by proofer** [jesseduffield/lazygit#2217] [Q&A] [answered]
I have two remotes, which lazygit seems to have picked up from .git/config, which is fine. They are `upstream` and `origin`. `origin` is my repo, and `upstream` is someone else's that I forked. I want to do all my pushes to `origin.` But lazygit wants to do them to `upstream`. I can't figure out how to change the push target.

**Comment by mark2185** [comment, marked as answer]
Are the pushes for all branches or just for one branch? 

If the former, you could just run a custom command through `:`, e.g. `:git push --all origin`.

If the latter, you can set the upstream of a branch by pressing `u` on it (`Local Branches` panel).

**Comment by aksh1618** [comment]
There's also [this custom command](https://github.com/jesseduffield/lazygit/wiki/Custom-Commands-Compendium#pushing-to-a-specific-remote-repository) that lets you choose the remote for the current push operation without having to set it as upstream

**Comment by rcdailey** [comment]
Late response but here's what I do:
1. I set `main` to track `upstream/main` so that when I do a pull or `Rebase onto base branch`, it always shows `upstream/main` instead of `origin/main`.
1. I do `:git config remote.pushDefault origin` so that when I push a branch, it goes to my fork. This lets the standard `ctrl+p` work without custom commands.
